lightkurve/lightcurvefile.py

```diff
@@ -265,9 +283,15 @@ def get_lightcurve(self, flux_type, centroid_type='MOM_CENTR'):
                 label=self.hdu[0].header['OBJECT'],
                 ra=self.ra,
                 dec=self.dec)
+        elif flux_type in ['SAP_FLUX', 'PDCSAP_FLUX']:
+            # Must be a custom FITS file:
+            warnings.warn("This custom file does not have a {} column. "
+                        "The available flux columns are {}".
+                        format(flux_type, self._flux_types()), LightkurveWarning)
+            return None # Could possibly return NaN array.
         else:
             raise KeyError("{} is not a valid flux type. Available types are: {}".
-                           format(flux_type, self._flux_types))
+                           format(flux_type, self._flux_types()))
```

Comment:
I think we can remove lines 286-291, because lines 292-294 already take care of producing an informative error message if the requested `flux_type` is not available.  (And I think it's better to raise a clear error here rather than returning `None`.)

Reply:
Currently `SAP_FLUX` appears as an attribute (`lcf.SAP_FLUX` exists), even when it is not available and would raise an error.  Is that OK?

Reply:
That doesn't worry me!  It would of course be better if the interact-files populated `SAP_FLUX` instead of `FLUX`, because the former is the conventional way of storing uncorrected aperture photometry flux. However that doesn't have to be resolved as part of this PR, and probably the real answer here is to start supporting/advertising `lcf.get_lightcurve()` as the preferred method of obtaining the default-choice light curve from a file.